src/main/kotlin/no/nav/syfo/application/ForesporselService.kt

```diff
@@ -1,12 +1,21 @@
 package no.nav.syfo.application
 
 import no.nav.syfo.domain.Foresporsel
+import no.nav.syfo.domain.Personident
 
-class ForesporselService(private val varselProducer: IVarselProducer) {
-    fun sendToNarmesteleder(foresporsel: Foresporsel): Result<Foresporsel> {
+class ForesporselService(
+    private val varselProducer: IVarselProducer,
+    private val repository: IForesporselRepository,
+) {
+    fun storeAndSendToNarmesteleder(foresporsel: Foresporsel): Result<Foresporsel> {
+        val storedForesporsel = repository.createForesporsel(foresporsel)
         return varselProducer.sendNarmesteLederVarsel(
-            foresporsel = foresporsel,
+            foresporsel = storedForesporsel,
```

Comment:
Hvis man alltid skal lagre og sende i samme sleng, trenger man da å endre publishedAt, eller kunne den bare blitt satt fra starten av? Alternativet er vel at POSTen bare lagrer, og så har man en jobb som publiserer til nærmeste leder og endrer `publishedAt`-feltet i databasen når den jobben har kjørt.

Reply:
Ja, det kan jo være vi trenger en cronjob for å forsøke å sende varsler på nytt (vi får jo ikke til å lage en distribuert transaksjon her, så vi må ta høyde for at lagring til db går bra, mens sending av varsel feiler). Men det går likevel an å forsøke å sende med en gang (men må legge på exception-håndtering).